h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
+
+    var getActiveElementIndex = () => {
+      return Array.prototype.slice.call(this._dropdownItems).
+        indexOf(element.querySelector('.js-search-bar-dropdown-menu-item--active'));
+    };
+
+    var clearActiveDropdownItem = () => {
```

Comment:
I think there's some inconsistent naming here: `getActiveElementIndex()` and then `clearActiveDropdownItem()`. Need to decide on "active element" vs "active dropdown item" and use the same one everywhere.